File: test/functional/IRDec/Zig/simple.zig
//
// RUN: zig build-obj -fllvm -fno-strip %s -femit-llvm-ir=%t.ll
// RUN: gluc %t.ll -print-interface | FileCheck -v %s
//
const std = @import("std");

// CHECK-DAG: func sayHello();
export fn sayHello() void {
    var threaded = std.Io.Threaded.init(std.heap.page_allocator, .{});
    defer threaded.deinit();
    std.Io.File.writeStreamingAll(
        .stdout(),
        threaded.io(),
        "Hello, World!\n"
    ) catch {};
}

// CHECK-DAG: func square(x: Int32) -> Int32;
export fn square(x: i32) i32 {
    return x * x;
}

// CHECK-DAG: func getCC() -> *UInt8;
export fn getCC() [*:0]const u8 {
    return "Zig";
}

File: test/functional/IRDec/Swift/simple.swift
//
// RUN: swiftc -parse-as-library -emit-ir -g -gdwarf-types %s -o %t.ll
// RUN: gluc %t.ll -print-interface | FileCheck -v %s
//

// CHECK: public func getPizzaCount() -> Int32;
public func getPizzaCount() -> Int32 {
    return 42
}

// CHECK: public func square(x: Int32) -> Int32;
public func square(x: Int32) -> Int32 {
    return x * x
}

// CHECK: public func printCC()
public func printCC() {
    print("Swift", terminator: "")
}
